src/reanimated2/animation/util.ts

```diff
@@ -179,8 +179,9 @@ function decorateAnimation<T extends AnimationObject | StyleLayoutAnimation>(
     let finished = true;
     tab.forEach((i, index) => {
       animation[i].current = RGBACurrent[index];
-      // @ts-ignore: disable-next-line
-      finished &= animation[i].onFrame(animation[i], timestamp);
+      const result = animation[i].onFrame(animation[i], timestamp);
+      // We really to assign this value to result, instead of passing it directly - otherwise once "finished" is false, onFrame won't be called
```

Comment:
```suggestion
      // We really need to assign this value to result instead of passing it directly - otherwise once "finished" is false, onFrame won't be called
```